### ISSUE TITLE
Question: How to do versioning for services

### ISSUE THREAD

**Issue by DerMambo** [meteorhacks/cluster#21]
Hi @arunoda,

if you have a lot of serivces and one changes from 1.0.0 to 2.0.0 including breaking "API" changes. How can I control that?

By naming conventions of the service?

    Cluster.register('ServiceA.v1')
    Cluster.register('ServiceA.v2')
    ...

Or would it be an idea to register like

    // register service
    Cluster.register('ServiceA', {version: '1.0.1'})

    // connect to service
    serviceConnection = Cluster.discoverConnection("mlServiceTest", {version: '1.0.1'});

**Comment by arunoda** [comment]
Right now, we don't have a way for now. Can't we simply change the service name?

**Comment by DerMambo** [comment]
Yes we could but I guess it would be nicer to have one service with versions instead of one service with multiple names?!

Also since the versioning is pretty standardized with semantic versioning we could benefit from that. I can imagine that a lot of tooling could be developed for a huge microservices architecture, like a micro-service-dependcy tool etc. So a standard way of versioning might be practical!?

If the versioning is optional it would also do no harm. Would you have a look to a PR if I make one? ;)

**Comment by arunoda** [comment]
I think I'm okay with that.
We'll let the versioning optional by default. (May be add 0.0.0) behind the scene.

I think, I like `Cluster.register('ServiceA', {version: '1.0.1'})` format. It's kind a verbose than the other.

**Comment by DerMambo** [comment]
All right. PR coming soon :)

**Comment by arunoda** [comment]
Awesome.

**Comment by rhyslbw** [comment]
:+1:

**Comment by DerMambo** [comment]
fyi: Soon will be little later since versioning is a bit more complicated since there are quite a lot of parts involved (proxy, store etc.)

**Comment by arunoda** [comment]
And I might have to change the store part a bit too. Don't worry. It just
placing things here and there. Wanted to add ETCD support.
